tests/getServiceProxy.test.js

```diff
@@ -73,3 +82,103 @@ tap.test('getServiceProxy available for testing', async t => {
   t.strictSame(result.payload.message, RETURN_MESSAGE)
   microserviceScope.done()
 })
+
+tap.test('getDirectServiceProxyFromUrlString available for testing', async t => {
+  nock.disableNetConnect()
+  t.tearDown(() => {
+    nock.enableNetConnect()
+  })
+
+  const RETURN_MESSAGE = 'OK'
+  const customProxy = getDirectServiceProxyFromUrlString(MY_AWESOME_SERVICE_PROXY_HTTP_URL)
+  const awesomeHttpServiceScope = nock(`${MY_AWESOME_SERVICE_PROXY_HTTP_URL}:80`)
+    .get('/test-endpoint')
+    .reply(200, {
+      message: RETURN_MESSAGE,
+    })
+
+  const result = await customProxy.get('/test-endpoint')
+
+  t.strictSame(result.statusCode, 200)
+  t.strictSame(result.payload.message, RETURN_MESSAGE)
+  awesomeHttpServiceScope.done()
+})
+
+tap.test('getDirectServiceProxyFromUrlString available for testing - https url', async t => {
+  nock.disableNetConnect()
+  t.tearDown(() => {
+    nock.enableNetConnect()
+  })
+
+  const RETURN_MESSAGE = 'OK'
+  const customProxy = getDirectServiceProxyFromUrlString(MY_AWESOME_SERVICE_PROXY_HTTPS_URL)
+  const awesomeHttpsServiceScope = nock(`${MY_AWESOME_SERVICE_PROXY_HTTPS_URL}:443`)
```

Comment:
I don't get why you want to add the port by means of interpolation; couldn't it be part of the constant `MY_AWESOME_SERVICE_PROXY_HTTPS_URL` in the first place? after all, the port is part of the URL.

Reply:
I did it just to be sure that the default ports are still working without specifying them.
I append the port into the nock only, but not into the `getDirectServiceProxyFromUrlString` function parameter.